src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -253,12 +253,17 @@ private void RemoveDocumentFromWorkspace_NoLock(MetadataAsSourceGeneratedFileInf
             }
         }
 
-        private void InitializeWorkspace(Project project)
+        private void InitializeWorkspace(Project project, OptionSet option)
```

Comment:
There's also more subtleties here I failed to consider when writing this code, consider bug #4206 that I just filed. That's more an edge case, but it's indicative of "MAS isn't passing options around right as it is".